component-test/test_dbus_proxy.py

```diff
@@ -92,162 +183,65 @@
-    }},
-    {{
-        "direction": "*",
-        "interface": "*",
-        "object-path": "*",
-        "method": "*"
-    }}],
-    "dbus-gateway-config-system": []
-}}
-""".format(**{
-    "iface": stubs.IFACE_1,
-    "extension_1": stubs.EXT_1
-})
-
-CONF_WHITELIST_PATH = """
-{{
-    "dbus-gateway-config-session": [{{
-        "direction": "*",
-        "interface": "*",
-        "object-path": "*",
-        "method": "*"
-    }},
+    """
+    CONF_CONTRADICTING_RULES = """
     {{
-        "direction": "incoming",
-        "interface": "{iface}.{extension_1}",
-        "object-path": "/a/path/to/unavailable/directory/",
-        "method": "*"
-    }}],
-    "dbus-gateway-config-system": []
-}}
-""".format(**{
-    "iface": stubs.IFACE_1,
-    "extension_1": stubs.EXT_1
-})
-
-CONF_WHITELIST_PATH_REVERSE = """
-{{
-    "dbus-gateway-config-session": [{{
-        "direction": "incoming",
-        "interface": "{iface}.{extension_1}",
-        "object-path": "/a/path/to/unavailable/directory/",
-        "method": "*"
-    }},
+        "dbus-gateway-config-session": [{{
+            "direction": "outgoing",
+            "interface": "{iface}.*",
+            "object-path": "*",
+            "method": "*"
+        }},
+        {{
+            "direction": "outgoing",
+            "interface": "{iface}.{extension_1}",
+            "object-path": "*",
+            "method": "*"
+        }}],
+        "dbus-gateway-config-system": []
+    }}
+    """.format(**{
+        "iface": stubs.IFACE_1,
+        "extension_1": stubs.EXT_1
+    })
+
+    """ Same as CONF_CONTRADICTING_RULES but a different order of the rules.
+        This is used to verify that the ordering of rules does not affect the
+        behavior.
+    """
+    CONF_CONTRADICTING_RULES_DIFFERENT_ORDER = """
     {{
-        "direction": "*",
-        "interface": "*",
-        "object-path": "*",
-        "method": "*"
-    }}],
-    "dbus-gateway-config-system": []
-}}
-""".format(**{
-    "iface": stubs.IFACE_1,
-    "extension_1": stubs.EXT_1
-})
-
-
-class TestWhitelisting(object):
+        "dbus-gateway-config-session": [{{
+            "direction": "outgoing",
+            "interface": "{iface}.{extension_1}",
+            "object-path": "*",
+            "method": "*"
+        }},
+        {{
+            "direction": "outgoing",
+            "interface": "{iface}.*",
+            "object-path": "*",
+            "method": "*"
+        }}],
+        "dbus-gateway-config-system": []
+    }}
+    """.format(**{
+        "iface": stubs.IFACE_1,
+        "extension_1": stubs.EXT_1
+    })
 
     @pytest.mark.parametrize("config", [
         CONF_WHITELIST_DIRECTIONS,
-        CONF_WHITELIST_DIRECTIONS_REVERSE
+        CONF_WHITELIST_DIRECTIONS_REVERSE,
+        CONF_WHITELIST_PATH,
+        CONF_WHITELIST_PATH_REVERSE,
+        CONF_WHITELIST_METHOD,
+        CONF_WHITELIST_METHOD_REVERSE,
+        CONF_CONTRADICTING_RULES,
+        CONF_CONTRADICTING_RULES_DIFFERENT_ORDER
```

Comment:
Please change the last two defines so they resemble the rest. Otherwise it's easy to wonder about if they are special somehow. In this case I guess it should be e.g. "CONF_WHITELIST_IFACE" etc.?